PortalFetch/crawler_sel.py

```diff
@@ -0,0 +1,332 @@
+    """
+    Get result contents from html.
+
+    Parameters: List, String
+    Returns: None
+    """
+    soup = BeautifulSoup(text, 'lxml')
+    trs = soup.find_all('tr')
+    for tr in trs:
+        ui = []
+        for td in tr:
+            ui.append(td.string)
+        ulist.append(ui)
+
+
+# input: urlist is the courses list we get from get_contents
+# output : save courses in a file "DeAnza2020spring.csv"
+def save_contents(filename, firstline, urlist):
+    """
+    Save course result to a .csv file.
+
+    Parameters: String, String, List
+    Returns: None
+    """
+    with open(filename, 'w', encoding='utf-8', newline='') as f:
+        writer = csv.writer(f)
+        writer.writerow([firstline])
+
+        for i in range(8,len(urlist)-1):
+            if len(urlist[i]) >=35:
+                writer.writerow([urlist[i][1], urlist[i][3], urlist[i][5],urlist[i][7], urlist[i][9], urlist[i][11],
+                            urlist[i][13], urlist[i][15], urlist[i][17],urlist[i][19], urlist[i][21], urlist[i][23],
+                            urlist[i][25], urlist[i][27], urlist[i][29],urlist[i][31], urlist[i][33], urlist[i][35]]
+                            )
+            else:
+                writer.writerow(urlist[i])
+    logging.info("Finished!")
+
+
+def main():
+    """Download course information from De Anza myportal."""
+    logging.basicConfig(filename='example.log', level=logging.INFO)
+    logging.debug('This message should go to the log file')
+    logging.info('So should this')
+    logging.warning('And this, too')
+    """Login De Anza portal."""
+    driver = webdriver.Chrome()
+    login_myportal(driver)
+    # The way to judge is that the left menu can be found in the interface, and the menu style has list-group-item)
+    while True:
+        try:
+            driver.find_element_by_class_name("list-group-item")
+        except NoSuchElementException:
+            time.sleep(1)
+            logging.info('wait for seconds...')
+            continue
+
+        break
+
+    # Course search page from homepage after login
+    openSearchPage(driver)
+
+    # choose course
+    selectelement = driver.find_element_by_tag_name("select")  # Because the page has only one drop-down box
+    # Select specified course
+    quarter_downlist = Select(selectelement)  # a Select object
+    quarter_downlist.select_by_value("202042")  # Element positioning through select objects, value positioning (value of option)
+    # search submit button
+    submitClick(driver)
+    # advance submit
+    advanceSearch(driver)
+
+    # Wait for a while to make sure the page loads
+    waitUtilPageLoaded(driver)
+    # Go to the advanced options page and start filling in various search terms
+    fillAdvanceSearch(driver)
+    # Save searched courses
+    html = saveResult(driver)  # see method
+    courseList = []
+    get_contents(courseList, html)  #Extract course information from html
+    filename ="DeAnza2020spring.csv"
```

Comment:
The `filename` could be changed later. For example, in the future, there might be 2020 fall, 2021 winter, 2021 spring, etc. Therefore, I would suggest to make it something we can configure without changing the Python file.